website/src/middleware.js

```diff
@@ -106,25 +107,25 @@ function isProxied(req) {
 }
 
 module.exports.forceSSL = function(req, res, next){
-  var baseUrl = nconf.get("BASE_URL");
-
   if(isHTTP(req) && !isProxied(req)) {
-    return res.redirect(baseUrl + req.url);
+    return res.redirect(BASE_URL + req.url);
   }
 
   next();
 }
 
 // Redirect to habitica for non-api urls
-// NOTE: Currently using a static 'habitica.com' string, rather than baseUrl,
-// to make rollback easy. Eventually, baseUrl should be migrated.
+// NOTE: Currently using a static 'habitica.com' string, rather than BASE_URL,
+// to make rollback easy. Eventually, BASE_URL should be migrated.
 
 function nonApiUrl(req) {
   return req.url.search(/\/api\//) === -1;
 }
 
 module.exports.forceHabitica = function(req, res, next) {
-  if(nconf.get('NODE_ENV') === 'production' && !isProxied(req) && nonApiUrl(req)) {
+  var isHabitRPG = req.headers.host === 'habitrpg.com';
+
+  if(IS_PROD && isHabitRPG && !isProxied(req) && nonApiUrl(req)) {
```

Comment:
We want to keep redirecting the majority of stuff ("example.habitrpg.com" correctly forwards to habitica.com, which it should). Could we look at whitelisting a bypass for the other domains instead?